src/lsdb/core/search/polygon_search.py

```diff
@@ -1,28 +1,48 @@
+from typing import List, Tuple
+
 import dask
+import healpy as hp
 import hipscat as hc
+import numpy as np
 import pandas as pd
-from spherical_geometry.polygon import SingleSphericalPolygon
+from hipscat.pixel_math.polygon_filter import CartesianCoordinates, SphericalCoordinates
+from lsst.sphgeom import ConvexPolygon, UnitVector3d
 
 
 @dask.delayed
-def polygon_filter(data_frame: pd.DataFrame, polygon: SingleSphericalPolygon, metadata: hc.catalog.Catalog):
+def polygon_filter(data_frame: pd.DataFrame, polygon: ConvexPolygon, metadata: hc.catalog.Catalog):
```

Comment:
How "heavy" is the `ConvexPolygon` object? Will it be a problem to hand the whole object between dask workers?

Reply:
Good question! It's hard to be sure but I think it shouldn't be a problem because this object is fully implemented in C++ under the hood and it only references the region class. I did some digging on how much an example polygon object with 4 to 6 vertices was weighing in memory on my machine and it seems to be under a steady 100 bytes.